src/django_components/attributes.py

```diff
@@ -0,0 +1,182 @@
+# Initial implementation based on attributes.py from django-web-components
+# See https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/templatetags/components.py
+# And https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/attributes.py
```

Comment:
As it says here, this file is mostly copied from django-web-components, with added types, and with a tweak here and there.